src/phoenix/datasets/dataset.py

```diff
@@ -175,15 +171,36 @@ def get_actual_score_column(self) -> "Union[Series[float]]":
     def _get_embedding_feature_column_names(
         self, embedding_feature_name: str
     ) -> EmbeddingColumnNames:
-        if self.schema.embedding_feature_column_names is None:
-            raise err.SchemaError(err.MissingField("embedding_feature_column_names"))
-        embedding_feature_column_names = self.schema.embedding_feature_column_names
-        if (
-            embedding_feature_name not in embedding_feature_column_names
-            or embedding_feature_column_names[embedding_feature_name] is None
-        ):
-            raise err.SchemaError(err.MissingEmbeddingFeatureColumnNames(embedding_feature_name))
-        return embedding_feature_column_names[embedding_feature_name]
+        if embedding_feature_name == "prompt":
+            if self.schema.prompt_column_names is None:
+                raise err.SchemaError(err.MissingField("prompt_column_names"))
+            return self._get_prompt_column_names()
+        elif embedding_feature_name == "response":
+            if self.schema.response_column_names is None:
+                raise err.SchemaError(err.MissingField("response_column_names"))
+            return self._get_response_column_names()
+        else:
+            if self.schema.embedding_feature_column_names is None:
+                raise err.SchemaError(err.MissingField("embedding_feature_column_names"))
+            embedding_feature_column_names = self.schema.embedding_feature_column_names
+            if (
+                embedding_feature_name not in embedding_feature_column_names
+                or embedding_feature_column_names[embedding_feature_name] is None
+            ):
+                raise err.SchemaError(
+                    err.MissingEmbeddingFeatureColumnNames(embedding_feature_name)
+                )
+            return embedding_feature_column_names[embedding_feature_name]
+
+    def _get_prompt_column_names(self) -> EmbeddingColumnNames:
+        if self.schema.prompt_column_names is None:
+            raise err.SchemaError(err.MissingField("prompt_column_names"))
+        return self.schema.prompt_column_names
+
+    def _get_response_column_names(self) -> EmbeddingColumnNames:
+        if self.schema.response_column_names is None:
+            raise err.SchemaError(err.MissingField("response_column_names"))
+        return self.schema.response_column_names
```

Comment:
`_get_embedding_feature_column_names` will look recognize the reserved embedding feature names (dictionary keys) "prompt" & "response" and have dedicated functions to search in the schema for the vector, raw data, and link to data fields.